### ISSUE TITLE
ENOENT: no such file or directory, open 'node:path'

### ISSUE THREAD

**Issue by locinus** [isaacs/path-walker#21]
Hi there !

TL;DR: when updating from 1.10.2 to 1.11.1, my Node.js tests fails with :

> ENOENT: no such file or directory, open 'node:path'

Node version: v18.12.0

----------------------------------

Complete story:

I'm using the Node.js NestJS framework and am maintaining a base npm package for my projects.

When releasing a new minor version of my base package yesterday - **which did not include any changes in the package dependencies** -, and upgrading it via `yarn upgrade my-package` on one of my project, now this project tests (via `yarn test`) breaks with "ENOENT: no such file or directory, open 'node:path'".

Now, I checked what changed in this project's `yarn.lock`, isolated each change one by one and these ones break it:

- glob@^10.3.10: 10.3.12 => 10.3.15
which now requires path-scurry "^1.11.0" instead of "^1.10.2"

- path-scurry@^1.10.2 (1.10.2) => path-scurry@^1.11.0 (1.11.1)

Any hint?

**Comment by rodikh** [comment]
+1

same story, Jest fails to handle `node:path`, and while `glob` version is locked on my side, internally it installs `path-scurry` with `^`.

- jest version: 27.0.2
- glob version: 10.2.6

**Comment by rodikh** [comment]
According to this: https://github.com/jestjs/jest/issues/11637
This was fixed after `jest@27.1.0`.

Looks like I can upgrade jest without any issue, and the error goes away.

**Comment by isaacs** [comment]
Node 18 supports `node:path` though. You weren't running node 18, perhaps?

**Comment by locinus** [comment]
Hi guys. Sorry about the long delay.

@isaacs: I indeed use Node 18, yes ?

We will try @rodikh solution by upgrading jest.

**Comment by isaacs** [comment]
@locinus no worries, oss happens at its own pace. please let me know if it fixes the problem. :)

**Comment by locinus** [comment]
I had validation from my team that an update to jest 27.1.0 solves the issue.
Many thanks!
